Sources/SwiftIfConfig/IfConfigEvaluation.swift

```diff
@@ -433,20 +434,77 @@ func evaluateIfConfig(
 }
 
 extension IfConfigClauseSyntax {
-  /// Determine whether this condition is "syntaxErrorsAllowed".
-  func syntaxErrorsAllowed(
-    configuration: some BuildConfiguration
+  /// Fold the operators within an #if condition, turning sequence expressions
+  /// involving the various allowed operators (&&, ||, !) into well-structured
+  /// binary operators.
+  public static func foldOperators(
+    _ condition: some ExprSyntaxProtocol
+  ) -> (folded: ExprSyntax, diagnostics: [Diagnostic]) {
```

Comment:
Did you intentionally make this method public? I don’t think it provides sufficient value to increase the API surface area of swift-syntax.

Reply:
https://github.com/swiftlang/swift-syntax/pull/2806 makes it internal